### PR TITLE
Docstring

### PR DESCRIPTION
**Pull request by xjjiang**
### Summary

This PR tries to complete docstrings based on [PEP 257 – Docstring Conventions](https://peps.python.org/pep-0257/) and [Style Guilde](https://numpydoc.readthedocs.io/en/latest/format.html#returns). I didn't try to complete the docstrings in unit tests. I also didn't try to complete the docstrings for standard methods like `setup()`, `compute()`, `compute_partials()`. 

`subsystems/aerodynamics/flops_based/aero_report.py` is removed. It was a post-mission step that didn't need to run during every phase. The component was a placeholder that did nothing so that wouldn't change any results.

`subsystems/aerodynamics/flops_based/mach_number.py` is removed. the `flight_conditions.py` file in atmosphere subsystem replaces it.

I ask everyone to read through and correct me if you find error.

### Related Issues

- Progress towards #371

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: aviary/examples/external_subsystems/simple_weight/simple_weight.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 
 class SimpleWeight(om.ExplicitComponent):
+    """A simple weight component that computes wing mass."""
```
Comment (Kenneth-T-Moore) [inline aviary/examples/external_subsystems/simple_weight/simple_weight.py:19]: This component is pretty fictional (wing mass being 1.5 times the engine mass is nonsensical).  Change to:

"A simple component that compute a wing mass as a function of the engine mass. That values are not representative of any existing aircraft, and the component is meant to demonstrate the concept of an externally calculated subsystem mass."

File: aviary/subsystems/propulsion/test/test_propeller_performance.py [hunk 2]
```diff
@@ -2,11 +2,9 @@
 
 import numpy as np
 import openmdao.api as om
-
```
Comment (Kenneth-T-Moore) [inline aviary/subsystems/propulsion/test/test_propeller_performance.py:5]: This test should have its docstrings converted back to comments.

File: aviary/visualization/dashboard.py [hunk 3]
```diff
@@ -213,6 +213,22 @@ def _dashboard_cmd(options, user_args):
...
+    from a JSON file.
+
+    Parameters
+    ----------
+    json_filepath : str
+        Path to the JSON file containing tabular data.
+
+    Returns
+    -------
+    pane : Panel.Pane or None
```
Comment (Kenneth-T-Moore) [inline aviary/visualization/dashboard.py:227]: We generally omit the return name in the Returns section, so this line would just be "Panel.Pane or None"

File: aviary/examples/external_subsystems/simple_weight/simple_weight.py (hunk 1 above)
Comment (xjjiang) [reply aviary/examples/external_subsystems/simple_weight/simple_weight.py]: Done.

File: aviary/visualization/dashboard.py (hunk 3 above)
Comment (xjjiang) [reply aviary/visualization/dashboard.py:227]: Okay. I will keep them.

File: aviary/subsystems/mass/flops_based/landing_gear.py [hunk 4]
```diff
@@ -250,6 +250,10 @@ def compute_partials(self, inputs, J):
 
 
 class NoseGearLength(om.ExplicitComponent):
+    """
+    Computation of landing gear OLEO length from main gear OLEO length:\n
+    NOSE_GEAR_OLEO_LENGTH = 0.7 * MAIN_GEAR_OLEO_LENGTH
```
Comment (jkirk5) [inline aviary/subsystems/mass/flops_based/landing_gear.py:255]:
```suggestion
    Computation of nose gear oleo strut length from main gear oleo strut length:
    NOSE_GEAR_OLEO_LENGTH = 0.7 * MAIN_GEAR_OLEO_LENGTH
```
Oleo is just the name for a specific kind of shock absorbing strut in landing gear, see **https://en.wikipedia.org/wiki/Oleo_strut**

File: aviary/subsystems/mass/flops_based/landing_group.py [hunk 5]
```diff
@@ -5,10 +5,15 @@
...
     LandingMass, LandingTakeoffMassRatio)
 from aviary.utils.aviary_values import AviaryValues
-from aviary.variable_info.variables import Aircraft, Mission
+from aviary.variable_info.variables import Aircraft
 
 
 class LandingMassGroup(om.Group):
+    """
+    Group of landing related components: LandingTakeoffMassRatio, MainGearLength,
+    NoseGearLength, LandingMass, etc.
```
Comment (jkirk5) [inline aviary/subsystems/mass/flops_based/landing_group.py:14]:
```suggestion
    Group of landing-related components for FLOPS-based mass subsystem:
    LandingTakeoffMassRatio, MainGearLength, NoseGearLength, LandingMass, etc.
```

File: aviary/subsystems/mass/flops_based/mass_premission.py [hunk 6]
```diff
@@ -50,6 +50,11 @@
 
 
 class MassPremission(om.Group):
+    """
+    Groups of mass components: CargoMass, TransportCargoContainersMass, TransportEngineCtrlsMass,
+    TransportAvionicsMass, FuelCapacityGroup, EngineMass, etc.
```
Comment (jkirk5) [inline aviary/subsystems/mass/flops_based/mass_premission.py:55]:
```suggestion
    Pre-mission group of top-level mass estimation groups and components for FLOPS-based analysis:
    CargoMass, TransportCargoContainersMass, TransportEngineCtrlsMass, TransportAvionicsMass,
    FuelCapacityGroup, EngineMass, etc.
```

**Review by jkirk5 — COMMENTED** [review]
Partial review

File: aviary/validation_cases/benchmark_tests/test_bench_multiengine.py [hunk 7]
```diff
@@ -36,6 +36,9 @@
 
 @use_tempdirs
 class MultiengineTestcase(unittest.TestCase):
+    """
+    Test the setup and run of a model with multi-engines.
```
Comment (jkirk5) [inline aviary/validation_cases/benchmark_tests/test_bench_multiengine.py:40]:
```suggestion
    Test the different throttle allocation methods for models with multiple, unique EngineModels.
```

File: aviary/variable_info/test/test_var_structure.py [hunk 8]
```diff
@@ -22,6 +23,10 @@ class OperatingLimits:
 
 
 class MetaDataTest(unittest.TestCase):
+    """
+    Test no duplicate FLOPS and LEAPS1 names and in alphabetic order. Also test no missing names.
```
Comment (jkirk5) [inline aviary/variable_info/test/test_var_structure.py:27]:
```suggestion
    Tests for variable_meta_data.py: check for duplicate legacy code names, alphabetization, and any missing names from variable hierarchy.
```

**Review by jkirk5 — CHANGES_REQUESTED** [review]
rest of review

File: aviary/subsystems/propulsion/test/test_propeller_performance.py (hunk 2 above)
Comment (xjjiang) [reply aviary/subsystems/propulsion/test/test_propeller_performance.py:5]: done.

File: aviary/examples/external_subsystems/battery/test_battery_builder.py [hunk 9]
```diff
@@ -8,6 +8,9 @@
 
 @av.skipIfMissingDependencies(BatteryBuilder)
 class TestBattery(av.TestSubsystemBuilderBase):
+    """
+    Test battery builder
+    """
```
[the remark this answers is not in the document]
Comment (xjjiang) [reply aviary/examples/external_subsystems/battery/test_battery_builder.py:13]: For a class, a docstring is okay when we run testflo.

File: aviary/examples/external_subsystems/engine_NPSS/test_NPSS_builder.py [hunk 10]
```diff
@@ -11,6 +11,9 @@
 
 
 class AviaryNPSSTestCase(unittest.TestCase):
+    """
+    Test NPSS engine builder from table by building an Aviary model with NPSS engine and run
+    """
```
[the remark this answers is not in the document]
Comment (xjjiang) [reply aviary/examples/external_subsystems/engine_NPSS/test_NPSS_builder.py:16]: Class docstring is fine with testflo run. I will keep it.

File: aviary/examples/external_subsystems/simple_weight/simple_weight.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (xjjiang) [reply aviary/examples/external_subsystems/simple_weight/simple_weight.py]: Done.

File: aviary/interface/methods_for_level2.py [hunk 11]
```diff
@@ -78,7 +77,14 @@
 
 
 class PreMissionGroup(om.Group):
+    """Pre mission group"""
```
[the remark this answers is not in the document]
Comment (xjjiang) [reply aviary/interface/methods_for_level2.py:80]: Why post-mission? I changed to "pre-mission".

File: aviary/subsystems/propulsion/engine_deck.py [hunk 12]
```diff
@@ -7,10 +7,6 @@
 
 Attributes
 ----------
-accepted_headers : dict
```
[the remark this answers is not in the document]
Comment (xjjiang) [reply aviary/subsystems/propulsion/engine_deck.py:10]: @jkirk5 What am I supposed to do here?

File: aviary/mission/gasp_based/phases/ascent_phase.py [hunk 13]
```diff
@@ -8,6 +8,22 @@
...
+    This class extends the PhaseBuilderBase class, providing specific implementations for
+    the ascent phase of a flight mission.
+
+    Attributes
+    ----------
+    Inherits all attributes from PhaseBuilderBase.
+
+    Methods
+    -------
+    Inherits all methods from PhaseBuilderBase.
```
[the remark this answers is not in the document]
Comment (xjjiang) [reply aviary/mission/gasp_based/phases/ascent_phase.py:23]: @jkirk5 So, what is the plan?

File: aviary/examples/external_subsystems/battery/battery_builder.py [hunk 14]
```diff
@@ -11,29 +11,30 @@ class BatteryBuilder(SubsystemBuilderBase):
...
-
     Attributes
     ----------
-    name : str ('battery_voltage')
+    name : str ('battery')
         object label
+    include_constraints: boolean
+        flag whether to include constraints.
 
     Methods
```
[the remark this answers is not in the document]
Comment (jkirk5) [reply aviary/examples/external_subsystems/battery/battery_builder.py:23]: I guess leave it for now. I don't know what the correct answer is

File: aviary/subsystems/propulsion/engine_deck.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (jkirk5) [reply aviary/subsystems/propulsion/engine_deck.py:10]: Keep the deleted lines but with the variable name updated to "aliases"

File: aviary/examples/external_subsystems/battery/battery_builder.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (xjjiang) [reply aviary/examples/external_subsystems/battery/battery_builder.py:23]: Sounds good.

File: aviary/subsystems/propulsion/engine_deck.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (xjjiang) [reply aviary/subsystems/propulsion/engine_deck.py:10]: Done.

**Review by jkirk5 — APPROVED** [review]

**Review by Kenneth-T-Moore — APPROVED** [review]

**Review by crecine — APPROVED** [review]
minor suggestions

**Pull request merged**